### ISSUE TITLE
require-automatic-updates doesn't understand when the condition is re-checked

### ISSUE THREAD

**Issue by Skalman** [eslint/eslint#12108]
**Tell us about your environment**

* **ESLint Version:** v6.1.0
* **Node Version:** v10.14.1
* **npm Version:** v6.7.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
  "parserOptions": {
    "ecmaVersion": 2018
  },
  "rules": {
    "require-atomic-updates": "error"
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
let i = 0;
async function f() {
	if (i) {
		await g();
		if (i) {
			i = 0; // ERROR HERE
		}
	}
}

```

```bash
npx eslint eslint-bug.js
```

**What did you expect to happen?**

No error. I'm re-checking the condition after the `await`, so now I'm sure that the variable contains what I expect it to contain.

**What actually happened? Please include the actual, raw output from ESLint.**

```
  6:4  error  Possible race condition: `i` might be reassigned based on an outdated value of `i`  require-atomic-updates

✖ 1 problem (1 error, 0 warnings)
```

**Are you willing to submit a pull request to fix this bug?** No
